crates/services/sync/src/import.rs

```diff
@@ -230,66 +245,58 @@ where
             }
         })
         .in_current_span()
-        .await
+        .await;
+
+        // Wait for any spawned tasks to shutdown
```

Comment:
You need to drop `shutdown_guard` before=)

Reply:
Normally, yes, but in this case `shutdown_guard` is consumed by:

```
let guard_stream = futures::stream::repeat_with(move || {
    (shutdown_guard.clone(), shutdown_signal.clone())
});
```

If I refactor to use your suggestion below, I would need to drop it manually.

Reply:
Actually, it seems like it gets consumed in both cases because I need to `move` capture with the `.map` anyway